tpgtools/api/apikeys/key.yaml

```diff
@@ -0,0 +1,242 @@
+                  x-dcl-list-type: list
+                  x-kubernetes-immutable: true
+              required:
+              - allowedBundleIds
+              type: object
+              x-dcl-go-name: IosKeyRestrictions
+              x-dcl-go-type: KeyRestrictionsIosKeyRestrictions
+            apiTargets:
+              description: A restriction for a specific service and optionally one or more specific methods.
+                Requests are allowed if they match any of these restrictions. If no restrictions are specified, all targets are allowed.
+              items:
+                properties:
+                  service:
+                    description: 'Required. The service for this restriction. 
+                      It should be the canonical service name, for example: translate.googleapis.com. You can use gcloud services list to get a list of services that are enabled in the project.'
+                    type: string
+                    x-dcl-go-name: Service
+                    # x-kubernetes-immutable: true
+                  methods:
+                    description: '
+                      Optional. List of one or more methods that can be called. 
+                      If empty, all methods for the service are allowed. A wildcard (*) can be used as the last symbol. Valid examples: google.cloud.translate.v2.TranslateService.GetSupportedLanguage TranslateText Get* translate.googleapis.com.Get*'
+                    items:
+                      type: string
+                      x-dcl-go-type: string
+                    type: array
+                    x-dcl-go-name: Methods
+                    x-dcl-list-type: list
+                    # x-kubernetes-immutable: true
+                required:
+                - attribute
+                - value
+                type: object
+                x-dcl-go-type: KeyRestrictionsAndroidKeyRestrictionsAllowedApplications
+              type: array
+              x-dcl-go-name: ApiTargets
+              x-dcl-list-type: list
+              x-kubernetes-immutable: true
+          type: object
+          x-dcl-go-name: Restrictions
+          x-dcl-go-type: KeyRestrictions
+        etag:
+          description: Output only. This checksum is computed by the server based
+            on the value of other fields, and may be sent only on create requests
+            to ensure the client has an up-to-date value before proceeding.
+          readOnly: true
+          type: string
+          x-dcl-go-name: Etag
+          x-kubernetes-immutable: true
+        name:
+          description: 'Output only. The resource name of the key.
+            The name has the form: projects/<PROJECT_NUMBER>/locations/global/keys/<KEY_ID>. For example: projects/123456867718/locations/global/keys/b7ff1f9f-8275-410a-94dd-3855ee9b5dd2'
+          type: string
+          x-dcl-go-name: Name
+          x-kubernetes-immutable: true
+          readOnly: true
```

Comment:
Did you copy this directly from the DCL source? It looks like this doesn't exist anymore: https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/apikeys/key.yaml#L55

Reply:
I wrote this by hand. I found that yaml file later while I was poking around the client internals. I tried that yaml file too and it returns the same error.

Reply:
Ah, yeah those files should be what we use to generate the TF resource. Ideally we will link directly to those, but I think that has been harder than expected.

From what I can see it seems like the DCL resource expects `name` to be defined during creation, but I'm confused because it says output only

Reply:
That is a bad assumption as some APIs won't have name field available until you do the 1st GET call after successful creation.

Reply:
Some recent APIs I worked on:

https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/create (Perfect example as the name field is a random serverside generated value) 


https://cloud.google.com/api-gateway/docs/reference/rest/v1beta/projects.locations.apis/create (Another example but the expected value that will be returned from the server is predictable)

Reply:
Yeah, I'm familiar with the server generated name field. It looks like this API uses the `keyId` specified in the create request as the ending of the name field, so it *should* be possible to remote `readOnly: true` on `name` and specify it during a test.

The DCL normal flow for creating a resource is to check for existence before creating, which is failing because name is empty.

There is certainly something wrong with the OpenAPI spec in this case, as name should probably be marked required or renamed to keyId or something similar, and I'll get that figured out.

Reply:
Cool, let me know when there is a fix available.

Reply:
Bump

Reply:
Should have a fix soon!

Reply:
Any update on this?